Sources/App/Commands/Analyze.swift

```diff
@@ -372,6 +372,12 @@ extension Analyze {
             .filter(\.$package.$id == pkgId)
             .all()
         let incoming = try await getIncomingVersions(client: client, logger: logger, package: package)
+        // TODO: temporary safe-guard
+        if !existing.isEmpty && incoming.isEmpty {
+            // Sudden loss of versions is suspicious, warn and throw error
+            logger.error("Suspicious loss of versions for package \(pkgId) - aborting analysis")
+            throw AppError.genericError(pkgId, "Suspicious loss of versions")
+        }
```

Comment:
This might not even be temporary. I've noticed the following error in our logs:

![CleanShot 2023-08-22 at 12 45 53@2x](https://github.com/SwiftPackageIndex/SwiftPackageIndex-Server/assets/65520/c59d36f9-036a-41d5-9d7c-0394d2dda68b)

If for some reason the shell `git tag` command returns just `"` with a status code 0, we'll fail to parse any tags and delete all versions. Our signal for failure is only the return code. I wonder if this is perhaps what's going wrong - we're getting nonsense from the command with a return code of 0.

Ideally we'd be doing our git stuff with a library to avoid this shell command hackery, but that's for another day.

It should be really rare and is highly likely a parsing error if a package loses all its releases. If we can't use a library to determine the tags, perhaps there are other ways to ensure we get all the tags. Perhaps the logs will tell us what exactly the shell is returning and solve this in the interim.